providers/task.rb

```diff
@@ -36,12 +36,22 @@
     validate_create_day
     validate_create_months
 
-    schedule = @new_resource.frequency == :on_logon ? 'ONLOGON' : @new_resource.frequency
+    schedule = case @new_resource.frequency
+               when :on_logon
+                 'ONLOGON'
+               when :on_idle
+                 'ONIDLE'
+               else
+                 @new_resource.frequency
+               end
     frequency_modifier_allowed = [:minute, :hourly, :daily, :weekly, :monthly]
     options = {}
     options['F'] = '' if @new_resource.force || task_need_update?
     options['SC'] = schedule
-    options['MO'] = @new_resource.frequency_modifier if frequency_modifier_allowed.include?(@new_resource.frequency)
+    unless @new_resource.frequency == :monthly && !%w(FIRST SECOND THIRD FOURTH LAST LASTDAY).include?(@new_resource.frequency_modifier)
+      options['MO'] = @new_resource.frequency_modifier if frequency_modifier_allowed.include?(@new_resource.frequency)
+    end
+    options['I']  = @new_resource.frequency_modifier if @new_resource.frequency == :on_idle
```

Comment:
might be a good idea just to add an attribute `idle_time` and not overload frequency_modifier.